vm/src/obj/objbytes.rs

```diff
@@ -155,8 +164,110 @@ impl PyBytesRef {
             iterated_obj: obj.into_object(),
         }
     }
+
+    fn add(self, other: PyObjectRef, vm: &VirtualMachine) -> PyObjectRef {
+        if let Ok(other) = other.downcast::<PyBytes>() {
+            let rhs = &other.value;
+            let elements: Vec<u8> = self.value.iter().chain(rhs.iter()).cloned().collect();
+            vm.ctx.new_bytes(elements)
+        } else {
+            vm.ctx.not_implemented()
+        }
+    }
+
+    fn contains(self, needle: PyObjectRef, vm: &VirtualMachine) -> PyObjectRef {
+        // no new style since objint is not.
+        if objtype::isinstance(&needle, &vm.ctx.bytes_type()) {
+            let result = vec_contains(&self.value, &get_value(&needle));
+            vm.ctx.new_bool(result)
+        } else if objtype::isinstance(&needle, &vm.ctx.int_type()) {
+            let result = self
+                .value
+                .contains(&objint::get_value(&needle).to_u8().unwrap());
+            vm.ctx.new_bool(result)
+        } else {
+            vm.new_type_error(format!("Cannot add {:?} and {:?}", self, needle))
+        }
+    }
+
+    fn getitem(self, needle: PyObjectRef, vm: &VirtualMachine) -> PyObjectRef {
+        if objtype::isinstance(&needle, &vm.ctx.int_type()) {
+            match objint::get_value(&needle).to_i32() {
+                Some(pos) => {
+                    if let Some(idx) = self.value.get_pos(pos) {
+                        vm.ctx.new_int(self.value[idx])
+                    } else {
+                        vm.new_index_error("index out of range".to_string())
+                    }
+                }
+                None => {
+                    vm.new_index_error("cannot fit 'int' into an index-sized integer".to_string())
+                }
+            }
+        } else if objtype::isinstance(&needle, &vm.ctx.slice_type()) {
+            vm.ctx
+                .new_bytes((self.value.get_slice_items(&vm, &needle)).unwrap())
+        } else {
+            vm.new_type_error(format!(
+                "byte indices must be integers or slices, not {}",
+                needle
+            ))
+        }
+    }
+
+    fn capitalize(self, vm: &VirtualMachine) -> PyObjectRef {
+        let mut new = vec![self.value[0].to_ascii_uppercase()];
+        new.extend(&self.value[1..]);
+        vm.ctx.new_bytes(new)
```

Comment:
Using `self.value[0]` here will panic if the `Vec` is empty. Also, you can just `.clone()` self and modify the first index using [`.get_mut()`](https://doc.rust-lang.org/std/vec/struct.Vec.html).